src/ORM/EagerLoadedList.php

```diff
@@ -268,17 +262,22 @@ public function addRows(array $rows): static
     /**
      * Not implemented - use addRow instead.
      */
-    public function add($item)
+    public function add(mixed $item): void
     {
         throw new BadMethodCallException('Cannot add a DataObject record to EagerLoadedList. Use addRow() to add database rows.');
     }
 
     /**
      * Removes a record from the list. Note that the record will not be removed from the
      * database - this list is read-only.
+     *
+     * @param DataObject $item
      */
-    public function remove($item): static
+    public function remove(mixed $item)
```

Comment:
The DataList subclasses are inconsistent with whether they return a DataObject or void.  DataList returns void, HasManyList returns void, ManyManyList returns a DataObject

The other SS_List implementations are inconsistent too. EagerLoadedList returns itself, ArrayList returns void

I feel that it's better to just leave the return type blank across the board here as that's what's most consistent

Reply:
We're adding explicit strong typing to these methods - we should consistently add it to the `remove()` method on _all_ classes.

There are two ways to do that:
1. adding the type that is currently used (`void` and `static` respectively depending on whether a value is returned)
2. consistently returning `$this` for all implementations of `remove()` and using `static` as the return typehint.

I'm not terribly fussed about which you choose to do (though I do prefer 2 because it gives a more consistent developer experience) - but we should do one of these.

> I feel that it's better to just leave the return type blank across the board here as that's what's most consistent

The most consistent thing to do would be to always return `$this` so if we're aiming for consistency lets go with 2 above.

Reply:
I don't want to change the the implemented return type remove() methods, particularly since EagerLoadedList returns static while ManyManyList returns a DataObject.  For this reason I want to keep the this as is.

I think that's out of scope for this PR. I'm trying my best to constrain scope in PRs these days :-)

While I've added strong typing to this PR, that's respecting existing implementations.

The only implementation changes this PR has made have been:
- Throwing exceptions if methods don't exist on the inner list on ListDecorator for - forTemplate(), columnUnique(), and debug() - as those methods aren't defined on SS_List
- Refactoring func_get_args() to ...$args, which is functionality identical

Reply:
That's fine, do 1 or 3 then.

Reply:
Feels bad that we are _removing_ strict types in this PR but whatever at this point.